app/common/directives/language-switch.directive.js

```diff
@@ -26,8 +26,10 @@ function LanguageSwitchController($scope, Languages, TranslationService) {
     }
 
     function changeLanguage(code) {
-        TranslationService.setLanguage(code);
-        TranslationService.translate(code);
+        if ($location.path() !== '/settings/general') {
```

Comment:
because before it was automatically changing the language when you changed it in settings and was not waiting for you to "save." Because they use the same template and directive for general settings and user language, I needed to stop the auto-translation when settings changed.